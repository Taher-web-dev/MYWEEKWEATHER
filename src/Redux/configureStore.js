import { createStore, combineReducers, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import cityReducer from './Position/Reducer/reducer';
import weatherReducer from './Weather/Reducer/reducer';
import toggleReducer from './Toggle/toggle';

const reducer = combineReducers({
  city: cityReducer,
  weather: weatherReducer,
  toggle: toggleReducer,
});
const store = createStore(
  reducer,
  applyMiddleware(thunk, logger),
);
export default store;
