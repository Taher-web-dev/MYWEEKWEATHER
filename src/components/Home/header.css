.MuiAutocomplete-root fieldset {
    border-color: #fff;
}

.MuiTextField-root fieldset {
  border-color: #fff;
}
