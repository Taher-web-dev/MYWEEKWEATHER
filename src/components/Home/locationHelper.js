import XMLHttpRequest from 'xhr2';
import { setCountry } from '../../Redux/Position/Action/action';

export const getCity = (coordinates, dispatch) => {
  const xhr = new XMLHttpRequest();
  const lat = coordinates[0];
  const lng = coordinates[1];
  function processRequest() {
    if (xhr.readyState === 4 && xhr.status === 200) {
      const response = JSON.parse(xhr.responseText);
      const loc = response.address;
      dispatch(setCountry(loc));
    }
  }
  xhr.open('GET', `https://us1.locationiq.com/v1/reverse.php?key=pk.f19e4fdca419f2e8ffa20180b18d27d6&lat=${
    lat}&lon=${lng}&format=json`, true);
  xhr.send();
  xhr.onreadystatechange = processRequest;
  xhr.addEventListener('readystatechange', processRequest, false);
};

export const getCurrentLocation = (setCoord, dispatch) => {
  navigator.geolocation.getCurrentPosition((position) => {
    const coords = [position.coords.latitude.toString(), position.coords.longitude.toString()];
    getCity(coords, dispatch);
    setCoord([position.coords.latitude, position.coords.longitude]);
  });
};

export const searchLatAndLngByStreet = async (location, setCoord) => {
  const url = `https://api.opencagedata.com/geocode/v1/json?q=${location}&key=59fbb7ff74d34f8486c9a37271339b21`;
  const result = await fetch(url);
  const res = await result.json();
  const pertinentResult = res.results[0];
  const { lat, lng } = pertinentResult.geometry;
  setCoord([lat, lng]);
};
