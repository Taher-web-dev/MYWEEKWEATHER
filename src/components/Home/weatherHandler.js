import SUNNY from '../../Statics/Images/sun.svg';
import CLOUDY from '../../Statics/Images/cloudy.png';
import RAINY from '../../Statics/Images/rainy-day.png';

export const weatherDescription = {
  clear: ['SUNNY', SUNNY],
  clouds: ['CLOUDY', CLOUDY],
  rain: ['RAINY', RAINY],
  searching: [null, null],
};

export const currentData = (weatherObj) => {
  let [description, wind, humidity, pressure, temp] = ['Searching', '', '', '', ''];
  try {
    const targetData = weatherObj.current;
    description = targetData.weather[0].main;
    wind = targetData.wind_speed;
    humidity = targetData.humidity;
    pressure = targetData.pressure;
    temp = targetData.temp;
  } catch {
    [description, wind, humidity, pressure, temp] = ['Searching', '', '', '', ''];
  }
  return [description, wind, humidity, pressure, temp];
};
export const currentDay = (dt) => {
  const days = ['SUN', 'MON', 'TUE', 'WED', 'THU', 'FRI', 'SAT'];
  const date = new Date(dt * 1000);
  const orderDay = date.getDay();
  return days[orderDay];
};
export const forecastData = (weatherObj) => {
  const forecastWeather = [];
  try {
    const dailies = weatherObj.daily;
    [1, 2, 3, 4, 5, 6].forEach((index) => {
      const dayWeather = dailies[index];
      const dayDt = dayWeather.dt;
      const description = dayWeather.weather[0].main;
      const temp = dayWeather.temp.day;
      const dayWeatherDescription = weatherDescription[description.toLowerCase()][0];
      const img = weatherDescription[description.toLowerCase()][1];
      const day = currentDay(dayDt);
      const dayWeatherObj = {
        temp, img, day, dayWeatherDescription,
      };
      forecastWeather.push(dayWeatherObj);
    });
  } catch {
    const forecastWeather = [];
    forecastWeather.push(1);
    forecastWeather.shift();
  }
  return forecastWeather;
};
