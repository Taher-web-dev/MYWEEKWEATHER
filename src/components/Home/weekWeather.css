.container {
  display: flex;
}

.weekday-weather {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.weather-img {
  margin-bottom: 20px;
  width: 40px;
}

.day-name {
  color: white;
  font-family: system-ui;
}
