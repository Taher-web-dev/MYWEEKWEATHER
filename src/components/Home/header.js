/* eslint-disable react/jsx-props-no-spreading */
import { React } from 'react';
import MenuIcon from '@mui/icons-material/Menu';
import Typography from '@mui/material/Typography';
import Box from '@mui/material/Box';
import TextField from '@mui/material/TextField';
import Autocomplete from '@mui/material/Autocomplete';
import { createTheme, ThemeProvider } from '@mui/material/styles';
import PropTypes from 'prop-types';
import Grid from '@mui/material/Grid';
import ArrowBackIosIcon from '@mui/icons-material/ArrowBackIos';
import './header.css';
import { useSelector, useDispatch } from 'react-redux';
import { setCountry } from '../../Redux/Position/Action/action';
import countries from './helper';
import LOCATION from '../../Statics/Images/location.png';
// import { lastDayOfDecade } from 'date-fns';

const Header = (props) => {
  const {
    getCurrentLocation,
  } = props;
  const city = useSelector((state) => state.city);
  const dispatch = useDispatch();
  const popUpOoptions = () => {
    const header = document.querySelector('.header');
    const options = document.querySelector('.options');
    header.style.display = 'none';
    options.style.display = 'flex';
  };

  const changeHeaderBar = () => {
    const header = document.querySelector('.header');
    const options = document.querySelector('.options');
    options.style.display = 'none';
    header.style.display = 'flex';
  };
  const countryHandlerChange = (e) => {
    setTimeout(changeHeaderBar, 1000);
    const country = e.target.textContent;
    dispatch(setCountry({
      country,
      city: '',
    }));
  };

  const theme = createTheme({
    palette: {
      primary: {
        main: '#fff',
      },
      secondary: {
        main: '#fff',
      },
      text: {
        primary: '#fff',
        secondary: '#fff',
      },
      action: {
        active: '#fff',
      },
      background: {
        paper: 'none',
      },
    },
  });

  return (
    <div>
      <Grid
        container
        spacing={0.5}
        className="options"
        style={{
          display: 'none', justifyContent: 'space-around', alignItems: 'center', paddingTop: '5%',
        }}
      >
        <ThemeProvider theme={theme}>
          <Grid item xs={2}>
            <ArrowBackIosIcon style={{ color: '#fff' }} onClick={changeHeaderBar} />
          </Grid>
          <Grid item xs={8}>
            <Autocomplete
              id="country-select-demo"
              sx={{ width: 190 }}
              options={countries}
              autoHighlight
              onChange={countryHandlerChange}
              getOptionLabel={(option) => option.label}
              renderOption={(props, option) => (
                <Box component="li" sx={{ '& > img': { mr: 2, flexShrink: 0 } }} {...props} style={{ color: '#fff' }}>
                  <img
                    loading="lazy"
                    width="20"
                    src={`https://flagcdn.com/w20/${option.code.toLowerCase()}.png`}
                    srcSet={`https://flagcdn.com/w40/${option.code.toLowerCase()}.png 2x`}
                    alt=""
                  />
                  {option.label}
                </Box>
              )}
              renderInput={(params) => (
                <TextField
                  {...params}
                  label="Choose a country"
                  inputProps={{
                    ...params.inputProps,
                    autoComplete: 'new-password', // disable autocomplete and autofill
                  }}
                />
              )}
            />
          </Grid>
        </ThemeProvider>
      </Grid>

      <div
        className="header"
        style={{
          display: 'flex', justifyContent: 'space-around', alignItems: 'center', paddingTop: '5%',
        }}
      >
        <MenuIcon style={{ color: '#fff' }} onClick={popUpOoptions} />
        <Typography variant="h5" component="div" style={{ color: '#fff' }}>
          {city.country}
          ,
          {city.city}
        </Typography>
        <div role="button" onClick={getCurrentLocation} onKeyDown={getCurrentLocation} tabIndex={0}>
          <img src={LOCATION} alt="location_icon" height="25rem" />
        </div>
      </div>
    </div>
  );
};
Header.propTypes = {
  getCurrentLocation: PropTypes.func.isRequired,
};
export default Header;
