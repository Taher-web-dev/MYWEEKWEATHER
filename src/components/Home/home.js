import { React, useState, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import XMLHttpRequest from 'xhr2';
import Grid from '@mui/material/Grid';
import Typography from '@mui/material/Typography';
import { setCountry } from '../../Redux/Position/Action/action';
import Header from './header';
import getWeatherData from '../../Redux/Weather/Thunk/thunk';
import { weatherDescription, currentData } from './weatherHandler';

const Home = () => {
  const [coord, setCoord] = useState([0, 0]);
  const dispatch = useDispatch();
  const weather = useSelector((state) => state.weather);
  const location = useSelector((state) => state.city);
  const toggle = useSelector((state) => state.toggle);
  const [
    currentDescription,
    currentWind,
    currentHumidity,
    currentPressure,
    currentTemp,
  ] = currentData(weather);
  const [
    currentWeatherDescription,
    currentImage,
  ] = weatherDescription[currentDescription.toLowerCase()];
  const getCity = (coordinates) => {
    const xhr = new XMLHttpRequest();
    const lat = coordinates[0];
    const lng = coordinates[1];
    function processRequest() {
      if (xhr.readyState === 4 && xhr.status === 200) {
        const response = JSON.parse(xhr.responseText);
        const loc = response.address;
        dispatch(setCountry(loc));
      }
    }
    xhr.open('GET', `https://us1.locationiq.com/v1/reverse.php?key=pk.f19e4fdca419f2e8ffa20180b18d27d6&lat=${
      lat}&lon=${lng}&format=json`, true);
    xhr.send();
    xhr.onreadystatechange = processRequest;
    xhr.addEventListener('readystatechange', processRequest, false);
  };

  const getCurrentLocation = () => {
    navigator.geolocation.getCurrentPosition((position) => {
      const coords = [position.coords.latitude.toString(), position.coords.longitude.toString()];
      getCity(coords);
      setCoord([position.coords.latitude, position.coords.longitude]);
    });
  };

  const weatherAnimation = () => {
    const element = document.querySelector('.weather');
    const translate = (ds) => {
      element.style.transform = `translate(${ds}px)`;
    };
    if (currentWeatherDescription) {
      if (currentWeatherDescription === 'SUNNY') {
        element.style.transform += 'rotate(1deg)';
        setInterval(weatherAnimation, 1);
      } else {
        translate(3);
        setTimeout(translate, 2000, 3);
        setTimeout(translate, 2000, -3);
        setTimeout(translate, 2000, -3);
        setTimeout(weatherAnimation, 3000);
      }
    }
  };

  const searchLatAndLngByStreet = async (location) => {
    const url = `https://api.opencagedata.com/geocode/v1/json?q=${location}&key=59fbb7ff74d34f8486c9a37271339b21`;
    const result = await fetch(url);
    const res = await result.json();
    const pertinentResult = res.results[0];
    const { lat, lng } = pertinentResult.geometry;
    setCoord([lat, lng]);
  };

  useEffect(() => getCurrentLocation(), []);
  useEffect(() => weatherAnimation(), [weather]);
  useEffect(() => dispatch(getWeatherData(coord[0], coord[1])), [coord]);
  useEffect(() => searchLatAndLngByStreet(location.country), [toggle]);
  return (
    <div>
      <Header getCurrentLocation={getCurrentLocation} />
      <Grid
        container
        style={{
          marginTop: '50px', backgroundColor: 'rgb(42, 84, 108)', padding: '25px 0', width: '90%', marginLeft: '5%', border: '4px solid rgb(96, 128, 148)', borderRadius: '5%',
        }}
      >
        <Grid
          item
          xs={6}
          style={{
            display: 'flex', flexDirection: 'column', alignItems: 'center', justifyContent: 'flex-end',
          }}
        >
          <img src={currentImage} alt="weather" style={{ marginBottom: '20px', width: '60px' }} className="weather" />
          <Typography variant="h5" style={{ color: '#fff' }}>{currentWeatherDescription}</Typography>
        </Grid>
        <Grid item xs={5} style={{ display: 'flex', flexDirection: 'column', justifyContent: 'flex-end' }}>
          <Typography variant="span" style={{ color: '#fff' }}>
            {`Wind: ${currentWind} m/s`}
          </Typography>
          <Typography variant="span" style={{ color: '#fff' }}>
            {`Humidity: ${currentHumidity} %`}
          </Typography>
          <Typography variant="span" style={{ color: '#fff' }}>
            {`Pressure: ${currentPressure} hPa`}
          </Typography>
          <Typography variant="h6" style={{ color: '#fff', textAlign: 'center' }}>
            {`${currentTemp}Â°c`}
          </Typography>
        </Grid>
      </Grid>
    </div>
  );
};
export default Home;
