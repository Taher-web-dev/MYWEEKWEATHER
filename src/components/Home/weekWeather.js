import { React } from 'react';
import Box from '@mui/material/Box';
import Typography from '@mui/material/Typography';
import Grid from '@mui/material/Grid';
import './weekWeather.css';
import PropTypes from 'prop-types';
import SUNNY from '../../Statics/Images/sun.svg';

const WeakWEather = (props) => {
  const { forecasting } = props;
  let index = 0;
  return (
    <Box sx={{ flexGrow: 1 }}>
      <Grid container spacing={{ xs: 2, md: 3 }} columns={{ xs: 6, sm: 12, md: 12 }} className="container">
        {Array.from(Array(6)).map(() => {
          index += 1;
          return (
            <Grid item xs={2} sm={4} md={4} key={index} className="weekday-weather">
              <Typography variant="h5" className="day-name"> MON </Typography>
              <img src={SUNNY} alt="weather-icon" className="weather-img" />
              <Typography variant="h6" style={{ color: '#fff', textAlign: 'center' }}>
                15Â°c
              </Typography>
            </Grid>
          );
        })}
        ;
      </Grid>
    </Box>
  );
};
WeakWEather.propTypes = {
  forecasting: PropTypes.instanceOf(Object).isRequired,
};
export default WeakWEather;
