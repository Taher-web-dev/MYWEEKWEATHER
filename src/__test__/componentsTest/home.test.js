import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Home from '../../components/Home/home';
import XMLHttpRequest from 'xhr2';

const mockStore = configureStore([]);
const mockGeolocation = {  getCurrentPosition: jest.fn(),  watchPosition: jest.fn()};global.navigator.geolocation = mockGeolocation;
describe('render correctly weekWeather component', () => {
  let store;
  let tree;
  beforeEach(() => {
    store = mockStore({
      city: {},
      weather: {},
      toggle: true,
    });
    store.dispatch = jest.fn();
    tree = render(
      <Provider store={store}>
        <Home />
      </Provider>,
    );
  });
  it('it should render home component with given state from redux store', () => {
    expect(tree).toMatchSnapshot();
  });
});
