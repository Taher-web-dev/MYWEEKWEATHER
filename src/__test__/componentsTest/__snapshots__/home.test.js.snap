// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly weekWeather component it should render home component with given state from redux store 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-0.5 options css-125mmfg-MuiGrid-root"
            style="display: none; justify-content: space-around; align-items: center; padding-top: 5%;"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1o7apob-MuiGrid-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ArrowBackIosIcon"
                focusable="false"
                style="color: rgb(255, 255, 255);"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11.67 3.87 9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
                />
              </svg>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 css-vok43g-MuiGrid-root"
            >
              <div
                aria-expanded="false"
                class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-xsu6ln-MuiAutocomplete-root"
                role="combobox"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-6yceho-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="false"
                    for="country-select-demo"
                    id="country-select-demo-label"
                  >
                    Choose a country
                  </label>
                  <div
                    class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-87bu8l-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-autocomplete="list"
                      aria-invalid="false"
                      autocapitalize="none"
                      autocomplete="new-password"
                      class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                      id="country-select-demo"
                      spellcheck="false"
                      type="text"
                      value=""
                    />
                    <div
                      class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
                    >
                      <button
                        aria-label="Open"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-rmvm5c-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                        tabindex="-1"
                        title="Open"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="ArrowDropDownIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M7 10l5 5 5-5z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-173wfxe"
                      >
                        <span>
                          Choose a country
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="header"
            style="display: flex; justify-content: space-around; align-items: center; padding-top: 5%;"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="MenuIcon"
              focusable="false"
              style="color: rgb(255, 255, 255);"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
            <div
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
              style="color: rgb(255, 255, 255);"
            >
              ,
            </div>
            <div
              role="button"
              tabindex="0"
            >
              <img
                alt="location_icon"
                height="25rem"
                src="location.png"
              />
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
          style="margin: 50px 5% 100px 5%; background-color: rgb(42, 84, 108); padding: 25px 0px; width: 90%; border: 4px solid rgb(96, 128, 148); border-radius: 5%;"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
            style="display: flex; flex-direction: column; align-items: center; justify-content: flex-end;"
          >
            <img
              alt="weather"
              class="weather"
              style="margin-bottom: 20px; width: 60px; transform: rotate(0deg);"
            />
            <h5
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
              style="color: rgb(255, 255, 255);"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-k3f0g4-MuiGrid-root"
            style="display: flex; flex-direction: column; justify-content: flex-end;"
          >
            <span
              class="MuiTypography-root MuiTypography-span css-37jdci-MuiTypography-root"
              style="color: rgb(255, 255, 255);"
            >
              Wind:  m/s
            </span>
            <span
              class="MuiTypography-root MuiTypography-span css-37jdci-MuiTypography-root"
              style="color: rgb(255, 255, 255);"
            >
              Humidity:  %
            </span>
            <span
              class="MuiTypography-root MuiTypography-span css-37jdci-MuiTypography-root"
              style="color: rgb(255, 255, 255);"
            >
              Pressure:  hPa
            </span>
            <h6
              class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
              style="color: rgb(255, 255, 255); text-align: center;"
            >
              Â°c
            </h6>
          </div>
        </div>
        <div
          class="MuiBox-root css-i9gxme"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-spacing-md-3 container css-1xgjpwf-MuiGrid-root"
          >
            ;
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-0.5 options css-125mmfg-MuiGrid-root"
          style="display: none; justify-content: space-around; align-items: center; padding-top: 5%;"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1o7apob-MuiGrid-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ArrowBackIosIcon"
              focusable="false"
              style="color: rgb(255, 255, 255);"
              viewBox="0 0 24 24"
            >
              <path
                d="M11.67 3.87 9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
              />
            </svg>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 css-vok43g-MuiGrid-root"
          >
            <div
              aria-expanded="false"
              class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-xsu6ln-MuiAutocomplete-root"
              role="combobox"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-6yceho-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="country-select-demo"
                  id="country-select-demo-label"
                >
                  Choose a country
                </label>
                <div
                  class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-87bu8l-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-autocomplete="list"
                    aria-invalid="false"
                    autocapitalize="none"
                    autocomplete="new-password"
                    class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                    id="country-select-demo"
                    spellcheck="false"
                    type="text"
                    value=""
                  />
                  <div
                    class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
                  >
                    <button
                      aria-label="Open"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-rmvm5c-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                      tabindex="-1"
                      title="Open"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="ArrowDropDownIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M7 10l5 5 5-5z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-173wfxe"
                    >
                      <span>
                        Choose a country
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="header"
          style="display: flex; justify-content: space-around; align-items: center; padding-top: 5%;"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="MenuIcon"
            focusable="false"
            style="color: rgb(255, 255, 255);"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
          <div
            class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            style="color: rgb(255, 255, 255);"
          >
            ,
          </div>
          <div
            role="button"
            tabindex="0"
          >
            <img
              alt="location_icon"
              height="25rem"
              src="location.png"
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
        style="margin: 50px 5% 100px 5%; background-color: rgb(42, 84, 108); padding: 25px 0px; width: 90%; border: 4px solid rgb(96, 128, 148); border-radius: 5%;"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
          style="display: flex; flex-direction: column; align-items: center; justify-content: flex-end;"
        >
          <img
            alt="weather"
            class="weather"
            style="margin-bottom: 20px; width: 60px; transform: rotate(0deg);"
          />
          <h5
            class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            style="color: rgb(255, 255, 255);"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-k3f0g4-MuiGrid-root"
          style="display: flex; flex-direction: column; justify-content: flex-end;"
        >
          <span
            class="MuiTypography-root MuiTypography-span css-37jdci-MuiTypography-root"
            style="color: rgb(255, 255, 255);"
          >
            Wind:  m/s
          </span>
          <span
            class="MuiTypography-root MuiTypography-span css-37jdci-MuiTypography-root"
            style="color: rgb(255, 255, 255);"
          >
            Humidity:  %
          </span>
          <span
            class="MuiTypography-root MuiTypography-span css-37jdci-MuiTypography-root"
            style="color: rgb(255, 255, 255);"
          >
            Pressure:  hPa
          </span>
          <h6
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
            style="color: rgb(255, 255, 255); text-align: center;"
          >
            Â°c
          </h6>
        </div>
      </div>
      <div
        class="MuiBox-root css-i9gxme"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-spacing-md-3 container css-1xgjpwf-MuiGrid-root"
        >
          ;
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
